app/tests/tests_10/test_collection_assets_endpoint.py

```diff
@@ -310,35 +310,46 @@ def setUp(self):  # pylint: disable=invalid-name
         self.client = Client(headers=get_auth_headers())
         self.maxDiff = None  # pylint: disable=invalid-name
 
-    def test_asset_endpoint_patch_put_href(self):
+    def test_asset_endpoint_patch_put_href_to_external_href(self):
         collection_name = self.collection['name']
         asset_name = self.asset['name']
         asset_sample = self.asset.copy()
 
+        path = f'/{STAC_BASE_V}/collections/{collection_name}/assets/{asset_name}'
+
+        #verify the asset is internal before patching it
+        response = self.client.get(path)
+        self.assertStatusCode(200, response)
+        self.assertFalse(response.json()['is_external'])
+
         put_payload = asset_sample.get_json('put')
-        put_payload['href'] = 'https://testserver/non-existing-asset'
-        patch_payload = {'href': 'https://testserver/non-existing-asset'}
+        put_payload['href'] = 'https://testserver/external-asset'
+        patch_payload = {'href': 'https://testserver/external-asset-2'}
 
-        path = f'/{STAC_BASE_V}/collections/{collection_name}/assets/{asset_name}'
         response = self.client.patch(path, data=patch_payload, content_type="application/json")
-        self.assertStatusCode(400, response)
-        description = response.json()['description']
-        self.assertIn('href', description, msg=f'Unexpected field error {description}')
-        self.assertEqual(
-            "Found read-only property in payload",
-            description['href'][0],
-            msg="Unexpected error message"
-        )
+        self.assertStatusCode(200, response)
 
         response = self.client.put(path, data=put_payload, content_type="application/json")
-        self.assertStatusCode(400, response)
-        description = response.json()['description']
-        self.assertIn('href', description, msg=f'Unexpected field error {description}')
-        self.assertEqual(
-            "Found read-only property in payload",
-            description['href'][0],
-            msg="Unexpected error message"
-        )
+        self.assertStatusCode(200, response)
+
+    def test_asset_endpoint_patch_put_external_href(self):
+        collection_name = self.collection['name']
+        asset_name = 'new-external-asset.tiff'
+        asset_sample = self.asset.copy()
```

Comment:
The test is not updating an existing external asset but creating a new asset and should not be part of this test class which is about updating. So you need to check in the setUp() method of the class if you already have an external asset and if not add one there, then in this test you take the external asset, first assert that it is external, then update it and assert that it has been updated.